cfme/v2v/conversion_hosts.py

```diff
@@ -0,0 +1,237 @@
+        vmware_ssh_key = TextInput(locator='.//textarea[@id="vmware-ssh-key-input"]')
+        osp_cert_switch = V2VBootstrapSwitch(label='Verify TLS Certificates for OpenStack')
+        osp_ca_cert = TextInput(locator='.//textarea[@id="openstack-ca-certs-input"]')
+
+        @property
+        def is_displayed(self):
+            return self.conv_host_key.is_displayed
+
+        def after_fill(self, was_change):
+            self.parent.configure_btn.click()
+
+    @View.nested
+    class results(View):  # noqa
+        msg = Text('.//h3[contains(@class,"blank-slate-pf-main-action")]')
+        close_btn = Button("Close")
+
+        @property
+        def is_displayed(self):
+            return self.msg.is_displayed
+
+
+@attr.s
+class ConversionHost(BaseEntity):
+    """Class representing conversion Host
+        Args:
+            target_provider: RHV or OSP
+            provider_name: rhv43 or osp13-ims
+            cluster: Default
+            hostname: host to be configured
+            conv_host_key: /etc/pki/ovirt-engine/keys/engine_id_rsa on RHV-M.
+            transformation_method: vddk or ssh
+            vddk_library_path: url for vddk
+            vmware_ssh_key: /var/lib/vdsm/.ssh/id_rsa on rhv host
+            osp_ca_cert : CA cert
+
+    """
+    target_provider = attr.ib()
+    provider_name = attr.ib()
+    cluster = attr.ib()
+    hostname = attr.ib()
+    conv_host_key = attr.ib()
+    transformation_method = attr.ib()
+    vddk_library_path = attr.ib(default=None)
+    vmware_ssh_key = attr.ib(default=None)
+    osp_cert_switch = attr.ib(default=None)
+    osp_ca_cert = attr.ib(default=None)
+
+
+@attr.s
+class ConversionHostCollection(BaseCollection):
+    """Collection object for conversion host object"""
+
+    ENTITY = ConversionHost
+
+    def create(self,
+               target_provider,
+               provider_name,
+               cluster,
+               hostname,
+               conv_host_key,
+               transformation_method,
+               vddk_library_path=None,
+               vmware_ssh_key=None,
+               osp_cert_switch=None,
+               osp_ca_cert=None):
+        """Configure new conversions host in UI
+        Args:
+            target_provider: RHV or OSP
+            provider_name: rhv43 or osp13-ims
+            cluster: Default
+            hostname: Hostname to be configured
+            conv_host_key: /etc/pki/ovirt-engine/keys/engine_id_rsa on RHV-M.
+            transformation_method: vddk or ssh
+            vddk_library_path: url for vddk
+            vmware_ssh_key: /var/lib/vdsm/.ssh/id_rsa on rhv host
+            osp_cert_switch: Yes or No
+            osp_ca_cert: Tls Cert
+
+        """
+        view = navigate_to(self, "Configure")
+        provider_type = ("Red Hat Virtualization" if target_provider.one_of(RHEVMProvider)
+                         else "Red Hat OpenStack Platform")
+        view.location.fill({"provider_type": provider_type,
```

Comment:
Thoughts on moving the `fill` method of the nested views into the nested views themselves? That way the create method can have just a single call to `view.fill`

Reply:
Each view is a tab and needs to be filled after waiting for the view .

Reply:
This tab view doesn't mean that what @john-dupuy is suggesting can't be used.

I would prefer to have a more consistent implementation here for filling this view.  First, `widgetastic_manageiq.WaitTab` will wait for `is_active` when the tab's child widgets are accessed.  Second, the WaitFillViewStrategy should be set on each tab if you want to have each widget in the tab to be displayed, or on the top level view to wait for every tab to be displayed before being filled.

Reply:
A `ConversionHost.as_fill_value()` method could provide the correct fill dictionary structure (with location, hosts, authentication keys) to match the view, so that you don't have to call fill multiple times here.  View fill behavior should be controlled by the view class, not by the Entity/Collection model.